cate/core/types.py

```diff
@@ -86,4 +92,146 @@ def format(cls, value: T) -> str:
         return str(value)
 
 
-# TODO (gailis, forman): add Like-derived types here...
\ No newline at end of file
+# ===== Like-derived types below =====
+class Variable(Like[List[str]]):
+    """
+    Type class for Variable selection objects
+
+    Accepts:
+        1. a string 'pattern1, pattern2, pattern3'
+        2. a list ['pattern1', 'pattern2', 'pattern3']
+
+    Converts to a list of strings
+    """
+    TYPE = Union[List[str], str]
+
+    @classmethod
+    def convert(cls, value: Any) -> List[str]:
+        """
+        Convert the given value to a list of variable name patterns.
+        """
+        if isinstance(value, str):
+            return(to_list(value))
+
+        if not isinstance(value, list):
+            raise ValueError('Variable name pattern can only be a string or a'
+                            ' list of strings.')
+
+        for item in value:
+            if not isinstance(item, str):
+                raise ValueError('Variable name pattern can only be a string'
+                                 ' or a list of strings.')
+
+        return value
+
+
+class PointLike(Like[Point]):
+    """
+    Type class for geometric Point objects
+
+    Accepts:
+        1. a Shapely Point
+        2. a string 'lon,lat'
+        3. a tuple (lon, lat)
+
+    Converts to a Shapely point
+    """
+    TYPE = Union[Point, str, Tuple[float, float]]
+
+    @classmethod
+    def convert(cls, value: Any) -> Point:
+        try:
+            if isinstance(value, Point):
+                return value
+            if isinstance(value, str):
+                pair = value.split(',')
+                return Point(float(pair[0]), float(pair[1]))
+            return Point(value[0], value[1])
+        except Exception:
+            raise ValueError('cannot convert value <%s> to %s' % (value, cls.name()))
+
+    @classmethod
+    def format(cls, value: Point) -> str:
+        return "%s, %s" % (value.x, value.y)
+
+class PolygonLike(Like[Polygon]):
+    """
+    Type class for geometric Polygon objects
+
+    Accepts:
+        1. a Shapely Polygon
+        2. a string 'min_lon, min_lat, max_lon, max_lat'
+        3. a WKT string 'POLYGON ...'
+        4. a list of coordinates [(lon, lat), (lon, lat), (lon, lat)]
+
+    Converts to a valid shapely Polygon.
+    """
+    TYPE = Union[Polygon, str, List[Tuple[float, float]]]
+
+    @classmethod
+    def convert(cls, value: Any) -> Polygon:
+        try:
+            if isinstance(value, Polygon):
+                if value.is_valid:
+                    return value
+            if isinstance(value, list):
+                polygon = Polygon(value)
+                if polygon.is_valid:
+                    return polygon
+            if isinstance(value, str):
+                value = value.lstrip()
+                if value[:7] == 'POLYGON':
```

Comment:
After a quick search it seems that according to the standard WKT is not case sensitive. Although pretty much always all caps are used. So this is a good point.